src/ray/protobuf/experimental/instance_manager.proto

```diff
@@ -135,6 +135,34 @@ message LaunchRequest {
   int64 request_ts_ms = 4;
 }
 
+// Termination request for an instance.
+message TerminationRequest {
+  enum Cause {
+    UNKNOWN = 0;
+    // Idle termination.
+    IDLE = 1;
+    // Per type max instance limit reached.
+    MAX_NUM_NODE_PER_TYPE = 2;
+    // Global max num nodes reached.
+    MAX_NUM_NODES = 3;
+  }
+  // id of the request.
+  string id = 1;
+  // The ray node id of the instance that's terminated.
+  string ray_node_id = 2;
+  // the instance id of the instance that's terminated.
+  string instance_id = 3;
+  // the reason of the termination.
+  Cause cause = 4;
+
+  // Idle time if the cause is IDLE.
+  optional int64 idle_time_ms = 5;
+  // The per type max instance limit if the cause is MAX_NUM_NODE_PER_TYPE.
+  optional int32 max_num_nodes_per_type = 6;
```

Comment:
I am not familiar with protobuf. Is there any reason to use `int` instead of `uint`?

Reply:
hmmm, not really, maybe unsigned int is better here.